tests/app/forms/test_questionnaire_form.py

```diff
@@ -1221,6 +1221,88 @@ def test_get_calculation_total_with_no_input(app, answer_store, list_store):
         ] % {"total": "10"}
 
 
+def test_sum_calculated_field_value_source_answer(app, answer_store, list_store):
+    answer_total = Answer(answer_id="total-answer", value=10)
+
+    answer_store.add_or_update(answer_total)
+
+    with app.test_request_context():
+        schema = load_schema_from_name("test_validation_sum_against_value_source")
+
+        question_schema = schema.get_block("breakdown-block").get("question")
+
+        form_data = MultiDict(
+            {
+                "breakdown-1": "",
+                "breakdown-2": "5",
+                "breakdown-3": "4",
+                "breakdown-4": "1",
+            }
+        )
+
+        expected_form_data = {
+            "csrf_token": None,
+            "breakdown-1": None,
+            "breakdown-2": Decimal("5"),
+            "breakdown-3": Decimal("4"),
+            "breakdown-4": Decimal("1"),
+        }
+        form = generate_form(
+            schema,
+            question_schema,
+            answer_store,
+            list_store,
+            metadata={},
+            response_metadata={},
+            form_data=form_data,
+        )
+
+        form.validate()
+        assert form.data == expected_form_data
```

Comment:
How does this assert test `sum_calculated_field_value_source_answer`. I was expecting to assert the value or validation errors?.

Reply:
It checks if the form passes the validation against the value source which is answers in this case.

Reply:
If you are checking it passes validation, then it should assert that a validation error is not raised.

Reply:
Doesn't seem to have changed. Is it committed?

Reply:
This one is addressed now.

Reply:
Not sure it is. https://github.com/ONSdigital/eq-questionnaire-runner/pull/882/files#r910129711

Reply:
We do:
```python 
if errors_text:
    for error_text in errors_text:
        assert (
            form.question_errors[question]
            == schema.error_messages[error_text] % value_dict
        )
else:
    assert len(form.question_errors) == 0
```
..so checking if validation errors are not raised?

Reply:
It does, if validation errors are raised it checks its contents. In `"schema_name, block, answers, breakdowns, expected_form_data, question, errors_text, value_dict",`  `error_text` name might be misleading, it is `validation error text`.

Reply:
If it raises a validator error, then that is what the first asset should be instead of `assert form.data == expected_form_data`, cause that doesn't really do anything.